webpack/hashed-folder-copy-plugin/src/HashedFolderDependency.ts

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import path from 'node:path';
+import crypto from 'node:crypto';
+import glob from 'fast-glob';
+import webpack from 'webpack';
+import type { SourceLocation } from 'estree';
+import { LegacyAdapters } from '@rushstack/node-core-library';
+
+import { renderError } from './HashedFolderCopyPlugin';
+import type {
+  ConnectionState,
+  DependencyTemplateContext,
+  ObjectDeserializerContext,
+  ObjectSerializerContext,
+  Range,
+  ResolverWithOptions,
+  UpdateHashContextDependency,
+  WebpackHash
+} from './webpackTypes';
+
+const makeSerializable: (
+  constructor: typeof HashedFolderDependency,
+  request: string
+) => void = require('webpack/lib/util/makeSerializable');
+
+export class HashedFolderDependencyTemplate extends webpack.dependencies.NullDependency.Template {
```

Comment:
`webpack@5` has various places that are really picky about object identity, so the safest way to get access to any internal components of webpack is via the `compiler.webpack` property.